pkg/catalog/list.go

```diff
@@ -0,0 +1,59 @@
+package catalog
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/weaveworks/pctl/pkg/subscription"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ProfileData data containing profile and available version update for format printing.
+type ProfileData struct {
+	Profile                 subscription.SubscriptionSummary
+	AvailableVersionUpdates []string
+}
+
+// List will fetch all installed profiles on the cluster and check if there are updated versions available.
+func List(k8sClient runtimeclient.Client, catalogClient CatalogClient) ([]ProfileData, error) {
+	profiles, err := subscription.NewManager(k8sClient).List()
+	if err != nil {
+		return nil, err
+	}
+	if len(profiles) == 0 {
+		fmt.Println("no profiles found")
+		return nil, nil
+	}
+	profileData := make([]ProfileData, 0)
+	for _, p := range profiles {
+		var versions []string
+		related, err := Search(catalogClient, p.Profile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to search for profile %s for updates: %w", p.Profile, err)
+		}
+		pv, err := version.NewVersion(strings.TrimPrefix(p.Version, "v"))
+		if err != nil {
+			return nil, fmt.Errorf("failed to format profile %s with version %s into version: %w", p.Profile, p.Version, err)
+		}
+		for _, r := range related {
+			// Search uses Contains, which will match too many things. We want an exact match.
+			// TODO: Maybe delegate this to the profile catalog?
+			if r.Name != p.Profile {
+				continue
+			}
+			rv, err := version.NewVersion(strings.TrimPrefix(r.Version, "v"))
+			if err != nil {
+				return nil, fmt.Errorf("failed to format profile %s with version %s into version: %w", r.Name, r.Version, err)
+			}
+			if rv.GreaterThan(pv) {
+				versions = append(versions, "v"+rv.String())
+			}
```

Comment:
does it make more sense for this logic to live in the profiles catalog API and to add a new endpoint? Similar to how the "latest" version logic all lives there too

Reply:
> // Search uses Contains, which will match too many things. We want an exact match.
> // TODO: Maybe delegate this to the profile catalog?

Yep. :D I agree with this question. :D I don't have the answer :) Does it? :D

Reply:
Oh so sorry. I still can't read apparently. You mean the version compare logic.
I guess? Maybe? I don't know, honestly. It would make the code more simple. It would just say, get GreaterThan and give it a profile name and current version. And then it would return all profiles and versions greater than.

Reply:
I STILL CAN'T READ. Because you mean 30-51. Frigging github for not highlighting highlighted rows for bigger comments.

Yes, I guess it would make sense to shift this whole logic into the catalog.

Reply:
:coffee: :heavy_check_mark: